### PR TITLE
Fix phase field not updating

### PR DESCRIPTION
**Pull request by gkrenn**
# Description
The oneagent reconciler references the wrong daemonset and therefore never updates the phase field.

## How can this be tested?
Deploy full stack monitoring and watch the phase field. => Should be "Running" after dynakube reconciliation.

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly

### CODE REVIEW

File: src/api/v1beta1/properties.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ const (
 	// PullSecretSuffix is the suffix appended to the DynaKube name to n.
 	PullSecretSuffix   = "-pull-secret"
 	TenantSecretSuffix = "-activegate-tenant-secret"
+
+	PodNameOSAgent = "oneagent"
```
Comment (mjgrzybek) [inline src/api/v1beta1/properties.go:35]:
```suggestion
	PodNameOsAgent = "oneagent"
```

File: src/controllers/dynakube/dynakube_controller_test.go [hunk 2]
```diff
@@ -221,7 +220,7 @@ func TestRemoveOneAgentDaemonset(t *testing.T) {
 			},
 			&appsv1.DaemonSet{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      testName + "-" + daemonset.PodNameOSAgent,
+					Name:      testName + "-" + dynatracev1beta1.PodNameOSAgent,
```
Comment (meik99) [inline src/controllers/dynakube/dynakube_controller_test.go:223]:
```suggestion
					Name:     instance.OneAgentDaemonsetName()
```

File: src/controllers/dynakube/dynakube_controller_test.go [hunk 3]
```diff
@@ -244,7 +243,7 @@ func TestRemoveOneAgentDaemonset(t *testing.T) {
 
 		var daemonSet appsv1.DaemonSet
 
-		err = controller.client.Get(context.TODO(), client.ObjectKey{Name: (testName + "-" + daemonset.PodNameOSAgent), Namespace: testNamespace}, &daemonSet)
+		err = controller.client.Get(context.TODO(), client.ObjectKey{Name: (testName + "-" + dynatracev1beta1.PodNameOSAgent), Namespace: testNamespace}, &daemonSet)
```
Comment (meik99) [inline src/controllers/dynakube/dynakube_controller_test.go:246]:
```suggestion
		err = controller.client.Get(context.TODO(), client.ObjectKey{Name: (instance.OneAgentDaemonsetName(), Namespace: testNamespace}, &daemonSet)
```

File: src/controllers/dynakube/oneagent/oneagent_reconciler_test.go [hunk 4]
```diff
@@ -92,10 +92,10 @@ func TestReconcileOneAgent_ReconcileOnEmptyEnvironmentAndDNSPolicy(t *testing.T)
 	assert.NoError(t, err)
 
 	dsActual := &appsv1.DaemonSet{}
-	err = fakeClient.Get(context.TODO(), types.NamespacedName{Name: dkName + "-" + daemonset.PodNameOSAgent, Namespace: namespace}, dsActual)
+	err = fakeClient.Get(context.TODO(), types.NamespacedName{Name: dkName + "-" + dynatracev1beta1.PodNameOSAgent, Namespace: namespace}, dsActual)
```
Comment (meik99) [inline src/controllers/dynakube/oneagent/oneagent_reconciler_test.go:95]:
```suggestion
	err = fakeClient.Get(context.TODO(), types.NamespacedName{Name: dynakube.OneAgentDaemonsetName(), Namespace: namespace}, dsActual)
```

File: src/controllers/dynakube/oneagent/oneagent_reconciler_test.go [hunk 5]
```diff
@@ -92,10 +92,10 @@ func TestReconcileOneAgent_ReconcileOnEmptyEnvironmentAndDNSPolicy(t *testing.T)
...
 	assert.NoError(t, err, "failed to get DaemonSet")
 	assert.Equal(t, namespace, dsActual.Namespace, "wrong namespace")
-	assert.Equal(t, dkName+"-"+daemonset.PodNameOSAgent, dsActual.GetObjectMeta().GetName(), "wrong name")
+	assert.Equal(t, dkName+"-"+dynatracev1beta1.PodNameOSAgent, dsActual.GetObjectMeta().GetName(), "wrong name")
```
Comment (meik99) [inline src/controllers/dynakube/oneagent/oneagent_reconciler_test.go:98]:
```suggestion
	assert.Equal(t,dynakube.OneAgentDaemonsetName(), dsActual.GetObjectMeta().GetName(), "wrong name")
```

File: src/api/v1beta1/properties.go [hunk 6]
```diff
@@ -63,6 +65,10 @@ func (dk *DynaKube) NeedsOneAgent() bool {
 	return dk.ClassicFullStackMode() || dk.CloudNativeFullstackMode() || dk.HostMonitoringMode()
 }
 
+func (dk *DynaKube) OneAgentDaemonsetName() string {
+	return fmt.Sprintf("%s-%s", dk.Name, PodNameOSAgent)
+}
```
Comment (meik99) [inline src/api/v1beta1/properties.go:70]: I'd suggest adding a unit test, which checks if this property is calculated correctly. Then the property can be used in other unit tests without having to manually concatenating it.

A very simple one would suffice. E.g.:
```
func TestOneAgentDaemonsetName(...) {
  instance := &DynaKube{
    ObjectMeta: {
      Name: "test",
    },
  }
  assert.Equal("test-oneagent", instance.OneAgentDaemonsetName())
}
```

**Review by meik99 — CHANGES_REQUESTED** [review]

File: src/testing/e2e/integrationtests/oneagent_controller_integration_test.go [hunk 7]
```diff
@@ -37,10 +36,10 @@ func TestReconcileOneAgent_ReconcileOnEmptyEnvironment(t *testing.T) {
 	// Check if deamonset has been created and has correct namespace and name.
 	dsActual := &appsv1.DaemonSet{}
 
-	err = e.Client.Get(context.TODO(), types.NamespacedName{Name: oaName + "-" + daemonset.PodNameOSAgent, Namespace: DefaultTestNamespace}, dsActual)
+	err = e.Client.Get(context.TODO(), types.NamespacedName{Name: oaName + "-" + dynatracev1beta1.PodNameOSAgent, Namespace: DefaultTestNamespace}, dsActual)
```
Comment (0sewa0) [inline src/testing/e2e/integrationtests/oneagent_controller_integration_test.go:39]: You could use `dynakube.OneAgentDaemonsetName()` if you change the inline creation of the dynakube in line 25 and put it into an actual variable

**Comment by meik99** [comment]
The phase field is updated initially while the deamonset is first deployed. However, if a oneagent pod is deleted after the phase has been set to running, it is not updated to read "Deploying" again. Is this expected behaviour?

File: src/api/v1beta1/properties.go (hunk 6 above)
Comment (gkrenn) [reply src/api/v1beta1/properties.go:70]: Ah, yeah, good idea. Haven't though about that. 👍

**Comment by gkrenn** [comment]
It should reset the phase to "Deploying" as the phase is determined on every oneagent reconcile which happens on every dynakube reconcile. 

The phase field in general is not implemented well, a more structured phase handling is done in the following PR: #609

**Review by 0sewa0 — APPROVED** [review]

**Review by meik99 — APPROVED** [review]

**Pull request merged**